### DISCUSSION TITLE
Writing 2D string dataset

### DISCUSSION THREAD

**Discussion by Murdius** [LiorBanai/HDF5-CSharp#319] [Q&A]
I'm trying to use https://github.com/LiorBanai/HDF5-CSharp/blob/main/HDF5-CSharp/Hdf5Dataset.cs#L300 to write a 2d string array. My code looks like:
```
long tef2 = Hdf5.CreateFile("asdf.H5");
string [,] blah = {{ "a", "b", "c" }};
Hdf5.WriteDatasetFromArray<string>(tef2, "blah", blah);
Hdf5.CloseFile(tef2)
```
But I'm getting an `ArgumentException: Object contains non-primitive or non-blittable data. (Parameter 'value')`

Is there some other way/trick to write a 2D string array into an HDF5 file?

**Comment by LiorBanai** [comment]
Hi @Murdius
I'll check this later this week and update you
